config-model-generator/src/main/java/io/strimzi/build/kafka/metadata/KafkaConfigModelGenerator.java

```diff
@@ -84,17 +84,34 @@ private static Map<String, ConfigModel> configs() throws NoSuchMethodException,
                 descriptor.setValues(enumer(key.validator));
             } else if (key.validator instanceof ConfigDef.ValidList) {
                 descriptor.setItems(validList(key));
-            } else if (key.validator instanceof ApiVersionValidator$) {
-                Iterator<ApiVersion> iterator = ApiVersion$.MODULE$.allVersions().iterator();
+            } else if (key.validator instanceof MetadataVersionValidator) {
```

Comment:
do you care if both MetadataVersion & Apiversions are set? do you wanna spot that out or check it out? 

I see here that it was actually totally removed so that's probably unlikely, right? 
apache/kafka#12072

Reply:
This is a code which is run against different Kafka versions. So it needs to handle changes like this where one version has `ApiVersion` and one has `MetadataVersion` But it is not completely trivial, because the code needs to be also compilable.